spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/partition/DeployerPartitionHandler.java

```diff
@@ -221,8 +221,11 @@ public void beforeTask(TaskExecution taskExecution) {
 		this.taskExecution = taskExecution;
 
 		if(this.commandLineArgsProvider == null) {
-			this.commandLineArgsProvider =
-					new SimpleCommandLineArgsProvider(this.taskExecution);
+			SimpleCommandLineArgsProvider provider = new
+					SimpleCommandLineArgsProvider();
+			provider.onTaskStartup(taskExecution);
```

Comment:
Why is this being created this way instead of using the listener correctly?  This implies that the deprication doesn't make sense in the first place...

Reply:
So should we just remove the deprecation?

Reply:
Why was it depricated?  We need to do the homework as to why it was.  I'd assume that we moved in this direction so that Spring would take care of the injection of the task execution.  If that's the case, shouldn't the provider be registered as a listener and injected here instead of newing up a new one?

Reply:
IMHO I think we should remove the deprecation itself and allow the Constructor to set the TaskExecution.  The SimpleCommandLineArgsProvider extends TaskExecutionListenerSupport that means that `onTaskStartup` its TaskExecution would get set.   However, SimpleCommandLineArgsProvider is never created as a bean and thus the `onTaskStartup` is never called.   And SimpleCommandLineArgsProvider is only used by the DeployerPartitionHandler, if a CommandLineArgsProvider has not been provided to the DeployerPartitionHandler.

Reply:
I don't have an argument against that so make it so :)